src/linear_algebra/finite_dimensional.lean

```diff
@@ -943,3 +955,77 @@ lemma set_is_basis_of_linear_independent_of_card_eq_findim
 is_basis_of_linear_independent_of_card_eq_findim lin_ind (trans s.to_finset_card.symm card_eq)
 
 end is_basis
+
+namespace module
+namespace End
+
+variables [finite_dimensional K V]
+
+lemma exists_ker_pow_eq_ker_pow_succ (f : End K V) :
+  ∃ (k : ℕ), k ≤ findim K V ∧ (f ^ k).ker = (f ^ k.succ).ker :=
+begin
+  classical,
+  by_contradiction h_contra,
+  simp_rw [not_exists, not_and] at h_contra,
+  have h_le_ker_pow : ∀ (n : ℕ), n ≤ (findim K V).succ → n ≤ findim K (f ^ n).ker,
+  { intros n hn,
+    induction n with n ih,
+    { exact zero_le (findim _ _) },
+    { have h_ker_lt_ker : (f ^ n).ker < (f ^ n.succ).ker,
+      { refine lt_of_le_of_ne _ (h_contra n (nat.le_of_succ_le_succ hn)),
+        rw pow_succ,
+        apply linear_map.ker_le_ker_comp },
+      have h_findim_lt_findim : findim K (f ^ n).ker < findim K (f ^ n.succ).ker,
+      { apply submodule.findim_lt_findim_of_lt h_ker_lt_ker },
+      calc
+        n.succ ≤ (findim K ↥(linear_map.ker (f ^ n))).succ :
+            nat.succ_le_succ (ih (nat.le_of_succ_le hn))
+        ... ≤ findim K ↥(linear_map.ker (f ^ n.succ)) :
+            nat.succ_le_of_lt h_findim_lt_findim } },
+  have h_le_findim_V : ∀ n, findim K (f ^ n).ker ≤ findim K V :=
+    λ n, submodule.findim_le _,
+  have h_any_n_lt: ∀ n, n ≤ (findim K V).succ → n ≤ findim K V :=
+    λ n hn, (h_le_ker_pow n hn).trans (h_le_findim_V n),
+  show false,
+    from nat.not_succ_le_self _ (h_any_n_lt (findim K V).succ (findim K V).succ.le_refl),
+end
+
+lemma ker_pow_constant {f : End K V} {k : ℕ} (h : (f ^ k).ker = (f ^ k.succ).ker) :
+  ∀ m, (f ^ k).ker = (f ^ (k + m)).ker
+| 0 := by simp
+| (m + 1) :=
+  begin
+    apply le_antisymm,
+    { rw [add_comm, pow_add],
+      apply linear_map.ker_le_ker_comp },
+    { rw [ker_pow_constant m, add_comm m 1, ←add_assoc, pow_add, pow_add f k m],
+      change linear_map.ker ((f ^ (k + 1)).comp (f ^ m)) ≤ linear_map.ker ((f ^ k).comp (f ^ m)),
+      rw [linear_map.ker_comp, linear_map.ker_comp, h, nat.add_one],
+      exact le_refl _, }
+  end
+
+lemma ker_pow_findim_add {f : End K V} {m : ℕ} :
+  (f ^ (findim K V + m)).ker = (f ^ findim K V).ker :=
```

Comment:
Why do you formulate it this way instead of considering `f^m` with `findim K V \leq m`?